sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/Statistics.scala

```diff
@@ -119,6 +120,18 @@ case class ColumnStat(
       maxLen = maxLen,
       histogram = histogram,
       version = version)
+
+  def updateCountStats(
+      oldNumRows: BigInt,
+      newNumRows: BigInt,
+      updatedColumnStatOpt: Option[ColumnStat] = None): ColumnStat = {
+    val updatedColumnStat = updatedColumnStatOpt.getOrElse(this)
+    val newDistinctCount = EstimationUtils.updateStat(oldNumRows, newNumRows,
+      distinctCount.get, updatedColumnStat.distinctCount.get)
```

Comment:
This looks a bit different from the previous code, which doesn't update if ndv is none

https://github.com/apache/spark/pull/33286/files#diff-44195511bd23aefa127db08fca79496fe99b2243c9262c5b3c62a6c8d37cff41L918